src/zoo.js

```diff
@@ -11,57 +11,137 @@ eslint no-unused-vars: [
 
 const data = require('./data');
 
-function animalsByIds(ids) {
-  // seu código aqui
-}
+const animalsByIds = (...ids) =>
+  data.animals.filter(animal => ids.find(id => id === animal.id));
 
-function animalsOlderThan(animal, age) {
-  // seu código aqui
-}
 
-function employeeByName(employeeName) {
-  // seu código aqui
-}
+const animalsOlderThan = (animal, age) =>
+  data.animals
+    .find(element => animal === element.name)
+    .residents.every(obj => obj.age >= age);
 
-function createEmployee(personalInfo, associatedWith) {
-  // seu código aqui
-}
+const employeeByName = function (name) {
+  return name ?
+    data.employees.find(element => element.firstName === name || element.lastName === name)
+    : {};
+};
+
+const createEmployee = (personalInfo, associatedWith) => ({ ...personalInfo, ...associatedWith });
+
+
+const isManager = id => data.employees.some(element => element.managers.includes(id));
+
+const addEmployee = (id, firstName, lastName, managers = [], responsibleFor = []) =>
+  data.employees.push(
+    { id, firstName, lastName, managers, responsibleFor });
+
+const animalCount = (species) => {
+  const bichos = {};
+  const filterAnimal = element => (bichos[element.name] = element.residents.length);
+  data.animals.filter(filterAnimal);
+  return (species) ? bichos[species] : bichos;
+};
+
+const entryCalculator = function (entrants) {
+  return !entrants || Object.keys(entrants).length === 0 ?
+    0
+    :
+    (data.prices.Adult * entrants.Adult) +
+    (data.prices.Child * entrants.Child) +
+    (data.prices.Senior * entrants.Senior);
+};
```

Comment:
Aqui da pra diminuir ali as linhas igual na outra, usando _array_ e _reduce_.
```suggestion
const entryCalculator = entrants =>
  (entrants && Object.keys(entrants).length > 0 ? Object.keys(entrants)
  .reduce((total, quantity) => (total + (data.prices[quantity] * entrants[quantity])), 0)
  : 0
);
```